### ISSUE TITLE
Maximum post/bio/username/display name lengths should be settings

### ISSUE THREAD

**Issue by ekiru** [glitch-soc/mastodon#144]
A lot of instances (and the fork itself) modify some of these lengths, but they have to do so in the source code, and it sometimes breaks tests that hard-code their assumptions about the lengths, as well as being unnecessary effort and making things trickier when merging.

We should make these into settings configurable in the admin interface.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by marrus-sh** [comment]
I've implemented this previously using environment variables; I agree that having it in the admin interface would be better but it sounds a little harder to pull off.

**Comment by marrus-sh** [comment]
I've added the `needs API` tag because all of this should be available through `initial_state` and `verify_credentials`

**Comment by clarcharr** [comment]
I agree that there should be configuration of these but also want to add another variable to this list: CWed post size.

This would allow admins to keep most post sizes small (e.g. 500 chars) but allow a larger size for posts with a CW (e.g. 10k chars).

**Comment by Pixeliz3d** [comment]
Maybe merge from https://github.com/florence-social/mastodon-fork ?

**Comment by ThibG** [comment]
Good idea!
I'm not 100% sure how to handle the transition from environment variables though.

**Comment by ashkitten** [comment]
perhaps we can add a migration that imports them into the database. main issue there would be that we need to make sure admins know about the migration, so they don't get confused why their environment variables aren't working anymore.

it's safe to leave the variables in .env, but we should consider removing them during migration to avoid confusion. this would break reverting the migration, though.

**Comment by leo60228** [comment]
This should probably be done for poll options too.
